src/terminal/kitty.rs

```diff
@@ -0,0 +1,328 @@
+// Copyright 2018-2019 Sebastian Wiesner <sebastian@swsnr.de>
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//  http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The kitty terminal.
+//!
+//! kitty is a fast, featureful, GPU based terminal emulator.
+//!
+//! See <https://sw.kovidgoyal.net/kitty/> for more information.
+
+use super::magic;
+use super::url::read_url;
+use failure::Error;
+use image::{ColorType, FilterType};
+use image::{DynamicImage, GenericImageView};
+use std::io::Write;
+use std::process::{Command, Stdio};
+use std::str;
+use url::Url;
+
+/// Whether we run in Kitty or not.
+pub fn is_kitty() -> bool {
+    std::env::var("TERM")
+        .map(|value| value == "xterm-kitty")
+        .unwrap_or(false)
+}
+
+/// Retrieve the terminal size by calling the command-line tool
+/// > kitty +kitten icat --print-window-size
+fn get_terminal_size_in_pixels() -> std::io::Result<(u32, u32)> {
```

Comment:
In order to resize the image to the terminal size I need the terminal size in *pixel*. The information of `Context.output.size` contains the number columns/rows instead of pixel. The API of crate `term_size` doesn't provide this information so I need to implement this function.

Reply:
I see :)